consensus/core/src/block_manager.rs

```diff
@@ -144,6 +144,53 @@ impl BlockManager {
         (accepted_blocks, missing_blocks)
     }
 
+    /// Tries to find the provided block_refs in DagState and adds any missing
+    /// blocks to the missing_blocks list
+    pub(crate) fn try_find_blocks(&mut self, mut block_refs: Vec<BlockRef>) -> BTreeSet<BlockRef> {
+        let _s = monitored_scope("BlockManager::try_find_blocks");
+
+        block_refs.sort_by_key(|b| b.round);
+        debug!(
+            "Trying to find blocks: {}",
+            block_refs.iter().map(|b| b.to_string()).join(",")
+        );
+
+        let mut missing_blocks = BTreeSet::new();
+
+        for (found, block_ref) in self
+            .dag_state
+            .read()
+            .contains_blocks(block_refs.clone())
+            .into_iter()
+            .zip(block_refs.iter())
+        {
+            let block_ref_hostname = &self.context.committee.authority(block_ref.author).hostname;
+            if !found {
```

Comment:
I believe we should extend this method a bit and we look into the suspended blocks as well. A block might has not been accepted by DagState but it might has been already fetched and sits in block manager as suspended hence there is no reason to attempt to fetch it.

Reply:
Good point, I will check `missing_blocks` as well as `suspended_blocks` in block manager and not duplicate fetch efforts if it is in either